File: src/main/java/com/ayssu/ciphergate/service/AppVariableService.java
package com.ayssu.ciphergate.service;

import com.ayssu.ciphergate.dto.AppVariableDTO;
import com.ayssu.ciphergate.dto.AppVariableQueryDTO;
import com.ayssu.ciphergate.entity.AppVariable;
import com.ayssu.ciphergate.entity.AppVariableHistory;
import com.baomidou.mybatisplus.extension.plugins.pagination.Page;

import java.util.List;
import java.util.Map;

/**
 * 应用变量服务接口
 */
public interface AppVariableService {
    
    /**
     * 分页查询变量
     */
    Page<AppVariable> getVariablePage(AppVariableQueryDTO queryDTO, Long operatorId);
    
    /**
     * 根据ID查询变量
     */
    AppVariable getVariableById(Long id, Long operatorId);
    
    /**
     * 根据应用ID和变量名称查询
     */
    AppVariable getVariableByName(Long appId, String variableName, Long operatorId);
    
    /**
     * 创建变量
     */
    AppVariable createVariable(AppVariableDTO dto, Long operatorId);
    
    /**
     * 更新变量
     */
    AppVariable updateVariable(Long id, AppVariableDTO dto, Long operatorId);
    
    /**
     * 删除变量
     */
    void deleteVariable(Long id, Long operatorId);
    
    /**
     * 批量删除变量
     */
    void deleteVariables(List<Long> ids, Long operatorId);
    
    /**
     * 复制变量
     */
    AppVariable copyVariable(Long id, String newVariableName, Long operatorId);
    
    /**
     * 获取应用变量（按变量名返回键值）
     */
    Map<String, Object> getAppVariables(Long appId, Long operatorId);
    
    /**
     * 批量更新变量值
     */
    void batchUpdateVariables(Long appId, Map<String, Object> variables, Long operatorId);
    
    /**
     * 获取变量历史记录
     */
    Page<AppVariableHistory> getVariableHistory(Long variableId, Integer current, Integer size, Long operatorId);
    
    /**
     * 导出变量配置
     */
    String exportVariables(Long appId, String format, Long operatorId);
    
    /**
     * 导入变量配置
     */
    void importVariables(Long appId, String configData, String format, Long operatorId);
    
    /**
     * 验证变量值
     */
    boolean validateVariableValue(AppVariable variable, String value);
    
}

File: frontend/src/services/appVariableService.ts
import request from '../utils/request';

export type VariableType = 'STRING' | 'NUMBER' | 'BOOLEAN' | 'JSON' | 'ARRAY';

export interface AppVariable {
  id: number;
  appId: number;
  appName?: string;
  variableName: string;
  displayName: string;
  description?: string;
  variableType: VariableType;
  variableValue?: string;
  required?: boolean;
  sortOrder?: number;
  validationRules?: string;
  options?: string;
  enabled?: boolean;
  version?: string;
  tags?: string;
  metadata?: Record<string, any>;
  createdBy?: number;
  updatedBy?: number;
  createdAt?: string;
  updatedAt?: string;
  deleted?: number;
}

export interface AppVariableQuery {
  appId?: number;
  variableName?: string;
  displayName?: string;
  variableType?: VariableType;
  enabled?: boolean;
  tag?: string;
  createdBy?: number;
  current?: number;
  size?: number;
}

export interface AppVariableDTO {
  id?: number;
  appId: number;
  variableName: string;
  displayName: string;
  description?: string;
  variableType: VariableType;
  variableValue?: string;
  required?: boolean;
  sortOrder?: number;
  validationRules?: string;
  options?: string;
  enabled?: boolean;
  version?: string;
  tags?: string;
  metadata?: Record<string, any>;
  changeReason?: string;
}

export interface AppVariableHistory {
  id: number;
  variableId: number;
  appId: number;
  variableName: string;
  operationType: 'CREATE' | 'UPDATE' | 'DELETE';
  oldValue?: string;
  newValue?: string;
  changeReason?: string;
  operatorId?: number;
  operatorIp?: string;
  operatedAt: string;
  version?: string;
}

/**
 * 分页查询变量
 */
export const getVariableList = (params: AppVariableQuery) => {
  return request.get('/app-variables', { params });
};

/**
 * 变量详情
 */
export const getVariableById = (id: number) => {
  return request.get(`/app-variables/${id}`);
};

/**
 * 根据名称查询变量
 */
export const getVariableByName = (appId: number, variableName: string) => {
  return request.get('/app-variables/by-name', { params: { appId, variableName } });
};

/**
 * 创建变量
 */
export const createVariable = (data: AppVariableDTO) => {
  return request.post('/app-variables', data);
};

/**
 * 更新变量
 */
export const updateVariable = (id: number, data: AppVariableDTO) => {
  return request.put(`/app-variables/${id}`, data);
};

/**
 * 删除变量
 */
export const deleteVariable = (id: number) => {
  return request.delete(`/app-variables/${id}`);
};

/**
 * 批量删除变量
 */
export const batchDeleteVariables = (ids: number[]) => {
  return request.delete('/app-variables/batch', { data: ids });
};

/**
 * 复制变量
 */
export const copyVariable = (id: number, newVariableName: string) => {
  return request.post(`/app-variables/${id}/copy`, { newVariableName });
};

/**
 * 获取变量历史记录
 */
export const getVariableHistory = (id: number, current = 1, size = 10) => {
  return request.get(`/app-variables/${id}/history`, { params: { current, size } });
};

/**
 * 导出应用变量配置
 */
export const exportAppVariables = (appId: number, params?: { format?: string }) => {
  return request.get(`/app-variables/app/${appId}/export`, { params });
};

/**
 * 导入应用变量配置
 */
export const importAppVariables = (
  appId: number,
  configData: string,
  params?: { format?: string }
) => {
  return request.post(`/app-variables/app/${appId}/import`, { configData }, { params });
};

/**
 * 验证变量值
 */
export const validateVariableValue = (id: number, value: string) => {
  return request.post(`/app-variables/${id}/validate`, { value });
};

